File: Dataset/Leetcode/train/62/237.java
 class Solution {
    public int XXX(int m, int n) {
        if(m == 1 || n == 1) return 1;
        m--;n--; // 调整使Start->Finish需要向下m格，向右n格
        long ans = m < n ? 
            myFactorial(m + n, n) / myFactorial(m, 0) : 
            myFactorial(m + n, m) / myFactorial(n, 0); 
        return (int)ans;
    }

    /** return c*(c-1)*...*(f+1) (ceil > floor)
     *  即 c!/f!
     */
    private long myFactorial(int ceil, int floor) {
        if (ceil <= floor) return 1;
        long ans = ceil--;
        // while (ceil > floor) ans *= ceil--;
        for (; ceil > floor; ceil--) ans *= ceil;
        return ans;
    }
}

File: Dataset/Leetcode/train/13/237.cpp
class Solution {
public:
    int XXX(string s) {
        int sum = 0;
        int size = s.size();
        unordered_map<char, int> mp;
        
        mp['I'] = 1;
        mp['V'] = 5;
        mp['X'] = 10;
        mp['L'] = 50;
        mp['C'] = 100;
        mp['D'] = 500;
        mp['M'] = 1000;
        
    
        for(int i = 0; i < size - 1; ++i) {
            if(mp[s[i]] >= mp[s[i + 1]])
                sum = sum + mp[s[i]];
            else
                sum = sum - mp[s[i]];
            }
        sum += mp[s[size - 1]];
        
        return sum;
    }
};

